include/mimalloc-internal.h

```diff
@@ -141,9 +144,17 @@ bool        _mi_page_is_valid(mi_page_t* page);
 // Overflow detecting multiply
 #define MI_MUL_NO_OVERFLOW ((size_t)1 << (4*sizeof(size_t)))  // sqrt(SIZE_MAX)
 static inline bool mi_mul_overflow(size_t size, size_t count, size_t* total) {
+#if __has_builtin(__builtin_umul_overflow) || __GNUC__ >= 5
+#if (MI_INTPTR_SIZE == 4)
+  return __builtin_umul_overflow(size, count, total);
+#else
+  return __builtin_umull_overflow(size, count, total);
+#endif
+#else /* non-GNU systems */
```

Comment:
Ah, sorry to nitpick: it works for clang too right? perhaps the comment should be /* no builtin umul_overflow */

Reply:
Yes, it works for clang as well. Thanks for pointing this.